src/unit/format.js

```diff
@@ -0,0 +1,49 @@
+define([
+	"../common/format-message"
+], function( formatMessage ) {
+
+/**
+ * format( value, unit, pluralGenerator )
+ *
+ * @value [Number]
+ *
+ * @unitProperies [Object]: localized unit data from cldr.
+ *
+ * @pluralGenerator [Object]: A pluralGenerator from Globalize.pluralGenerator.
+ *
+ * TODO pass along numberFormatter
```

Comment:
Since the unit formatter needs to access the actual number, passing a custom number formatter sounds fine to me. Would be good to include in this PR.